packages/cache/src/internal/cacheHttpClient.ts

```diff
@@ -113,6 +114,21 @@ export async function getCacheEntry(
   const cacheResult = response.result
   const cacheDownloadUrl = cacheResult?.archiveLocation
   if (!cacheDownloadUrl) {
+    // List cache for primary key only if cache miss occurs
+    const resource = `caches?key=${encodeURIComponent(keys[0])}`
```

Comment:
It is debatable if we should use cache key or the last of the restore keys. If restore keys are specified by user then it means we did not find a matching cache even with restore key. So should we restrict to only cache key for showing candidate caches or use the least restrictive restore key.
One problem with least restrictive restore key would that the list of candidate caches can be reallly long and make the logs too verbose.

Reply:
I think we should stick to primary key as cache miss means miss on primary key. Even if cache is found for restore keys it is still said cache not hit